kafka-observer/cmd/consumer/main.go

```diff
@@ -0,0 +1,155 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/IBM/CodeEngine/kafka-observer/internal/cmd"
+	"github.com/IBM/sarama"
+)
+
+var (
+	version = ""
+	oldest  = true
+	verbose = false
+)
+
+var idleTimer *time.Timer
+
+// Consumer implements the ConsumerGroupHandler
+// interface
+type Consumer struct {
+	ready chan bool
+}
+
+func main() {
+	fmt.Println("retrieving config")
+
+	config := cmd.GetConfig()
+
+	keepRunning := true
+	log.Println("Starting a new Sarama consumer")
+
+	if verbose {
+		sarama.Logger = log.New(os.Stdout, "[sarama] ", log.LstdFlags)
+	}
+
+	version = sarama.DefaultVersion.String()
+	version, err := sarama.ParseKafkaVersion(version)
+	if err != nil {
+		log.Panicf("Error parsing Kafka version: %v", err)
+	}
+
+	saramaConfig := sarama.NewConfig()
+	saramaConfig.Version = version
+	saramaConfig.Consumer.Offsets.AutoCommit.Enable = true
+	saramaConfig.ClientID, _ = os.Hostname()
+	saramaConfig.Net.SASL.Enable = true
+	saramaConfig.Net.SASL.User = config.KafkaAuth.User
+	saramaConfig.Net.SASL.Password = config.KafkaAuth.Token
+	saramaConfig.Net.TLS.Enable = true
+
+	if oldest {
+		saramaConfig.Consumer.Offsets.Initial = sarama.OffsetOldest
+	}
+
+	saramaConfig.Consumer.Group.Rebalance.GroupStrategies = []sarama.BalanceStrategy{sarama.NewBalanceStrategyRoundRobin()}
+
+	consumer := Consumer{
+		ready: make(chan bool),
+	}
+
+	brokers := config.Kafka.Brokers
+	topics := strings.Split(os.Getenv("KAFKA_TOPIC"), ",")
+
+	ctx, cancel := context.WithCancel(context.Background())
+	client, err := sarama.NewConsumerGroup(brokers, "consuming-group", saramaConfig)
```

Comment:
Unlike the run.sh script indicates, the consumer group is not configurable in the consumer. See `run.sh`
```
--env CONSUMER_GROUP=payment-consumer-group
```